src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -407,10 +418,39 @@ void CheckIsValidBranchTarget(BasicBlock src, BasicBlock target)
                     else
                         VerificationError(VerifierError.BranchOutOfHandler);
                 }
-                return;
             }
         }
 
+        bool IsDirectChildRegion(BasicBlock enclosingBlock, BasicBlock enclosedBlock)
+        {
+            int enclosingRegionStart;
+            if (enclosingBlock.TryIndex == null)
+                enclosingRegionStart = enclosingBlock.StartOffset;
+            else
+                enclosingRegionStart = _exceptionRegions[(int)enclosingBlock.TryIndex].ILRegion.TryOffset;
+
+            var enclosedRegion = _exceptionRegions[(int)enclosedBlock.TryIndex].ILRegion;
+
+            // Walk from enclosed start backwards and check each BasicBlock whether it is a try-start
+            for (int i = enclosedBlock.StartOffset - 1; i > enclosingRegionStart; --i)
```

Comment:
Would walking from `enclosedRegion.TryOffset - 1` simplify the logic below?

Reply:
Yes, this would reduce the blocks that need to be checked and makes the same instruction check performed below redundant.